internal/pkg/manifest/validate.go

```diff
@@ -588,6 +589,24 @@ func (s StaticSiteConfig) validate() error {
 }
 
 func (f FileUpload) validate() error {
+	if err := f.validateSource(f.Source); err != nil {
+		return err
+	}
+	return nil
+}
+
+// validateSource returns nil if Source is configured correctly.
+func (f FileUpload) validateSource(source string) error {
+	if source == "" {
+		return &errFieldMustBeSpecified{
+			missingField: "source",
+		}
+	}
+	fs := afero.NewOsFs()
+	_, err := fs.Stat(f.Source)
+	if err != nil {
+		return fmt.Errorf("source '%s' must be a valid path", f.Source)
+	}
```

Comment:
Can we guarantee that the `err` can always be attributed to the path being invalid? 

In addition, small nit:
```suggestion
	if err != nil {
		return fmt.Errorf("source '%s' must be a valid path: %w", f.Source, err)
	}
```